Let helloworld example listen to all hosts

thallgren opened the pull request:
This PR changes the address that the helloworld greeter_server is
listening to from "localhost:<port>" to just ":<port>", thus enabling
the server to be used in setups where the request stem from another
host.

RELEASE NOTES: N/A

easwars commented:
Why are the changes in the `pb.go` files required? They seem to be changing the protoc version to an older one. We recently changes all these files in https://github.com/grpc/grpc-go/pull/4038.

thallgren commented:
The `pg.go` changes were necessary to get a green `make all`. It updates all the`pb.go` files automatically. `vet.sh` then fails unless they are checked in. I'm not able to run a green `make all` on the current master. It fails here:

```
+ git status
On branch master
Your branch is up to date with 'origin/master'.

Changes not staged for commit:
  (use "git add <file>..." to update what will be committed)
  (use "git restore <file>..." to discard changes in working directory)
        modified:   balancer/grpclb/grpc_lb_v1/load_balancer.pb.go
        modified:   balancer/grpclb/grpc_lb_v1/load_balancer_grpc.pb.go
        modified:   binarylog/grpc_binarylog_v1/binarylog.pb.go
        modified:   channelz/grpc_channelz_v1/channelz.pb.go
        modified:   channelz/grpc_channelz_v1/channelz_grpc.pb.go
        modified:   credentials/alts/internal/proto/grpc_gcp/altscontext.pb.go
        modified:   credentials/alts/internal/proto/grpc_gcp/handshaker.pb.go
        modified:   credentials/alts/internal/proto/grpc_gcp/handshaker_grpc.pb.go
        modified:   credentials/alts/internal/proto/grpc_gcp/transport_security_common.pb.go
        modified:   examples/features/proto/echo/echo.pb.go
        modified:   examples/features/proto/echo/echo_grpc.pb.go
        modified:   examples/helloworld/helloworld/helloworld.pb.go
        modified:   examples/helloworld/helloworld/helloworld_grpc.pb.go
        modified:   examples/route_guide/routeguide/route_guide.pb.go
        modified:   examples/route_guide/routeguide/route_guide_grpc.pb.go
        modified:   health/grpc_health_v1/health.pb.go
        modified:   health/grpc_health_v1/health_grpc.pb.go
        modified:   internal/proto/grpc_lookup_v1/rls.pb.go
        modified:   internal/proto/grpc_lookup_v1/rls_config.pb.go
        modified:   internal/proto/grpc_lookup_v1/rls_grpc.pb.go
        modified:   internal/proto/grpc_service_config/service_config.pb.go
        modified:   interop/grpc_testing/benchmark_service.pb.go
        modified:   interop/grpc_testing/benchmark_service_grpc.pb.go
        modified:   interop/grpc_testing/control.pb.go
        modified:   interop/grpc_testing/core/stats.pb.go
        modified:   interop/grpc_testing/empty.pb.go
        modified:   interop/grpc_testing/messages.pb.go
        modified:   interop/grpc_testing/payloads.pb.go
        modified:   interop/grpc_testing/report_qps_scenario_service.pb.go
        modified:   interop/grpc_testing/report_qps_scenario_service_grpc.pb.go
        modified:   interop/grpc_testing/stats.pb.go
        modified:   interop/grpc_testing/test.pb.go
        modified:   interop/grpc_testing/test_grpc.pb.go
        modified:   interop/grpc_testing/worker_service.pb.go
        modified:   interop/grpc_testing/worker_service_grpc.pb.go
        modified:   profiling/proto/service.pb.go
        modified:   profiling/proto/service_grpc.pb.go
        modified:   reflection/grpc_reflection_v1alpha/reflection.pb.go
        modified:   reflection/grpc_reflection_v1alpha/reflection_grpc.pb.go
        modified:   reflection/grpc_testing/proto2.pb.go
        modified:   reflection/grpc_testing/proto2_ext.pb.go
        modified:   reflection/grpc_testing/proto2_ext2.pb.go
        modified:   reflection/grpc_testing/test.pb.go
        modified:   reflection/grpc_testing/test_grpc.pb.go
        modified:   stress/grpc_testing/metrics.pb.go
        modified:   stress/grpc_testing/metrics_grpc.pb.go
        modified:   test/codec_perf/perf.pb.go
        modified:   test/grpc_testing/test.pb.go
        modified:   test/grpc_testing/test_grpc.pb.go

no changes added to commit (use "git add" and/or "git commit -a")
+ git --no-pager diff
diff --git a/balancer/grpclb/grpc_lb_v1/load_balancer.pb.go b/balancer/grpclb/grpc_lb_v1/load_balancer.pb.go
index c393d7ff..70fb8fed 100644
--- a/balancer/grpclb/grpc_lb_v1/load_balancer.pb.go
+++ b/balancer/grpclb/grpc_lb_v1/load_balancer.pb.go
@@ -20,7 +20,7 @@
 // Code generated by protoc-gen-go. DO NOT EDIT.
 // versions:
 //     protoc-gen-go v1.25.0
-//     protoc        v3.14.0
+//     protoc        v3.13.0
 // source: grpc/lb/v1/load_balancer.proto
 ... # long list with diff of all the .pb.go files
  package grpc_testing
+ exit 1
+ cleanup
+ git reset --hard HEAD
HEAD is now at 344b93a2 testdata: use SHA256 as signing algorithm in testdata certs (#5074)
make: *** [Makefile:33: vet] Error 1
```

thallgren commented:
I used `make all` because your [contributing guidelines](/grpc/grpc-go/blob/master/CONTRIBUTING.md) recommends that in order to test everything. Using `go test` works.

menghanl commented:
Can you revert the changes to the `pb.go` files? I think only the port change is needed. Thanks.

thallgren commented:
I have removed that commit.
